### DISCUSSION TITLE
Feature Request or Help.

### DISCUSSION THREAD

**Discussion by muhannad-hash** [laurentS/slowapi#221] [Q&A]
**Describe the bug**
I have the following route in my fastapi app `/search` which accept a `search_term: str` and `data: bool`
if the data flag is set to true I am returnin some files from gcp related to the search done by the user so It's adding overcost to the search.

I want to rate limit the users only when they set the `data=True` and the response return more than 1 hit.
I would appreciate your help how can I do so.??!!
I am not sure wether this is available in your current implementation, or something you are interested in adding it.

Thanks a lot.

**Comment by laurentS** [comment]
Hi @muhannad-hash I think [this issue](https://github.com/laurentS/slowapi/issues/13) should help with your use case.
